Branch-SDK/Branch-SDK/Networking/BNCNetworkService.m

```diff
@@ -0,0 +1,279 @@
+        configuration.timeoutIntervalForResource = self.defaultTimeoutInterval;
+        configuration.URLCache = nil;
+
+        self.sessionQueue = [NSOperationQueue new];
+        self.sessionQueue.name = @"io.branch.network.queue";
+        self.sessionQueue.maxConcurrentOperationCount = self.maximumConcurrentOperations;
+        self.sessionQueue.qualityOfService = NSQualityOfServiceUserInteractive;
+
+        _session =
+            [NSURLSession sessionWithConfiguration:configuration
+                delegate:self
+                delegateQueue:self.sessionQueue];
+        _session.sessionDescription = @"io.branch.network.session";
+
+        return _session;
+    }
+}
+
+- (void) setSuspendOperations:(BOOL)suspendOperations {
+    self.sessionQueue.suspended = suspendOperations;
+}
+
+- (BOOL) operationsAreSuspended {
+    return self.sessionQueue.isSuspended;
+}
+
+#pragma mark - Operations
+
+- (BNCNetworkOperation*) networkOperationWithURLRequest:(NSMutableURLRequest*)request
+                completion:(void (^)(BNCNetworkOperation*operation))completion {
+
+    BNCNetworkOperation *operation = [BNCNetworkOperation new];
+    operation.request = request;
+    operation.networkService = self;
+    operation.completionBlock = completion;
+    return operation;
+}
+
+- (void) startOperation:(BNCNetworkOperation*)operation {
+    operation.networkService = self;
+    if (!operation.dateStart)
+        operation.dateStart = [NSDate date];
+    if (!operation.timeoutDate) {
+        operation.timeoutDate =
+            [[operation dateStart] dateByAddingTimeInterval:self.defaultTimeoutInterval];
+    }
+    operation.request.timeoutInterval = [operation.timeoutDate timeIntervalSinceDate:[NSDate date]];
+    operation.sessionTask =
+        [self.session dataTaskWithRequest:operation.request
+            completionHandler:
+            ^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+                operation.responseData = data;
+                operation.response = (NSHTTPURLResponse*) response;
+                operation.error = error;
+                operation.dateFinish = [NSDate date];
+                BNCLogDebug(@"Network finish operation %@ %1.3fs. Status %ld error %@.\n%@.",
+                    operation.request.URL.absoluteString,
+                    [operation.dateFinish timeIntervalSinceDate:operation.dateStart],
+                    (long)operation.response.statusCode,
+                    operation.error,
+                    operation.stringFromResponseData);
+                if (operation.completionBlock)
+                    operation.completionBlock(operation);
+            }];
+    BNCLogDebug(@"Network start operation %@.", operation.request.URL);
+    [operation.sessionTask resume];
+}
+
+- (void) cancelAllOperations {
+    [self.session invalidateAndCancel];
+    _session = nil;
+}
+
+- (void) URLSession:(NSURLSession *)session
```

Comment:
Just for the sake of cleanliness, can you remove the HTTPS cert pinning from this PR? I don't think we should convolute these two changes. It also makes this harder to review.